tests/test_server.py

```diff
@@ -210,32 +219,117 @@ def test_get_wal_full_path(self, tmpdir):
         full_path = server.get_wal_full_path(wal_name)
         assert full_path == str(tmpdir.join("wals").join(wal_hash).join(wal_name))
 
+    @pytest.mark.parametrize(
+        "wal_info_files,expected_indices",
+        [
+            (
+                [
+                    create_fake_info_file("000000010000000000000003", 42, 43),
+                    create_fake_info_file("00000001.history", 42, 43),
+                    create_fake_info_file("000000020000000000000003", 42, 43),
+                    create_fake_info_file("00000002.history", 42, 43),
+                    create_fake_info_file("000000030000000000000005", 42, 43),
+                    create_fake_info_file("00000003.history", 42, 43),
+                ],
+                [1, 2, 3, 5],
+            ),
+            (
+                [
+                    create_fake_info_file("000000010000000000000003", 42, 43),
+                    create_fake_info_file("00000001.history", 42, 43),
+                    create_fake_info_file("000000020000000000000003", 42, 43),
+                    create_fake_info_file("000000020000000000000010", 42, 43),
+                    create_fake_info_file("00000002.history", 42, 43),
+                    create_fake_info_file("000000030000000000000005", 42, 43),
+                    create_fake_info_file("00000003.history", 42, 43),
+                ],
+                [1, 2, 3, 4, 6],
+            ),
+        ],
+    )
+    def test_get_required_xlog_files(self, wal_info_files, expected_indices, tmpdir):
+        # Prepare input string
+        walstring = ""
+        for i in wal_info_files:
+            walstring += i.to_xlogdb_line()
+        # Prepare expected list
+        expected_wals = []
+        for j in expected_indices:
+            expected_wals.append(wal_info_files[j].name)
+
+        # create a xlog.db and add those entries
+        wals_dir = tmpdir.mkdir("wals")
+        xlog = wals_dir.join("xlog.db")
+        xlog.write(walstring)
+        # fake backup
+        backup = build_test_backup_info(
+            begin_wal="000000020000000000000001", end_wal="000000020000000000000004"
+        )
+
+        # mock a server object and mock a return call to get_next_backup method
+        server = build_real_server(
+            global_conf={"barman_lock_directory": tmpdir.mkdir("lock").strpath},
+            main_conf={"wals_directory": wals_dir.strpath},
+        )
+
+        wals = []
+        for wal_file in server.get_required_xlog_files(backup, 2, 41):
+            # get the result of the xlogdb read
+            wals.append(wal_file.name)
+        # Check for the presence of expected files
+        assert expected_wals == wals
```

Comment:
It took me a little while to figure out that we are verifying that get_required_xlog_files returns all WALs on the target timeline along with the history files - it could be helpful for future maintainers if the test explained this somewhere, though I'm not sure where best. Maybe in the parameterize function or the test docstring?